### PR TITLE
Reformat readme and add table of contents

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/setup-jfrog-cli#build-the-code) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

https://github.com/yahavi/setup-jfrog-cli/blob/add-toc/README.md

* Give the README a makeover
* Include a table of contents
* Streamline the OIDC instructions:
  * Keep it super basic - focus on getting it to function for a given repository
  * Drop the nitty-gritty details (like core.getIDToken())
  * Ditch the links to external docs; they clash a bit with this Action
* Enhance the OIDC test a tad.

### CODE REVIEW

File: .github/workflows/oidc-test.yml [hunk 1]
```diff
@@ -33,42 +33,32 @@ jobs:
...
         shell: bash
         run: |
           curl -X POST "${{ secrets.JFROG_PLATFORM_URL }}/access/api/v1/oidc" -H "Content-Type: application/json" -H "Authorization: Bearer ${{ secrets.JFROG_PLATFORM_RT_TOKEN }}" -d '{
-          "name": "${{ env.OIDC_CONFIG_NAME }}",
+          "name": "${{ env.OIDC_PROVIDER_NAME }}",
           "issuer_url": "https://token.actions.githubusercontent.com/",
           "provider_type": "GitHub",
           "description": "This is a test configuration created for OIDC-Access integration test" }'
 
-      - name: Set subject
```
Comment (eranturgeman) [inline .github/workflows/oidc-test.yml:47]: Can you please explain to me what does this step suppose to do?

File: README.md [hunk 2]
```diff
@@ -59,198 +79,217 @@ You can use these environment variables in your workflow as follows:
...
 - run: |
-    # Set the JFrog configuration to use by providing the server ID (configured by the 'jf c add' command).
-    jf c use local-1
-    # Ping local-1 Artifactory server
-    jf rt ping
-    # Now use the second sever configuration exposed to the Action.
-    jf c use local-2
-    # Ping local-2 Artifactory server
-    jf rt ping
+      # Set the JFrog configuration to use by providing the server ID (configured by the 'jf c add' command).
```
Comment (eranturgeman) [inline README.md:116]: Maybe rephrase for better understanding:
Set the utilized JFrog configuration by providing the server ID (configured by the 'jf c add' command).

**Review by eranturgeman — COMMENTED** [review]
The updated documentation looks great and the idea to add pictures is very helpful

File: README.md [hunk 3]
```diff
@@ -9,46 +9,66 @@
...
 
-# Usage
-## General
+-   Two distinct methods are available for authenticating with the JFrog Platform. Explore more details [here](#authorization)
+-   There's no need to add the _build name_ and _build number_ options and arguments to commands which accept them.
+    All build related operations will be automatically recorded with the _Workflow Name_ as build name and _Run Number_ as build number.
+
+## Usage
+
+### General
```
Comment (eyalbe4) [inline README.md:38]: This **General** seems redundant.

File: README.md [hunk 4]
```diff
@@ -9,46 +9,66 @@
...
+
+## Authorization
+
 JFrog CLI operates in conjunction with the JFrog Platform. In order to facilitate this connection, certain connection details of the JFrog Platform must be provided.
 There exist two methods to provide these details, and you only need to choose **one** method:
-* The connection details of the JFrog platform can be stored as secrets. Read more about it [here](#storing-jfrog-connection-details-as-secrets).
-* The connection details of the JFrog platform can be auto-generated using OpenID Connect protocol. Read more about it [here](#connect-using-openid-connect).
 
+<details>
+    <summary>Storing JFrog connection details as secrets</summary>
```
Comment (eyalbe4) [inline README.md:51]: Storing JFrog connection details as secrets
-->
Connecting to JFrog using secrets

File: README.md [hunk 5]
```diff
@@ -59,198 +79,217 @@ You can use these environment variables in your workflow as follows:
...
 
-## Connect using OpenID Connect
-The sensitive connection details, such as the access token used by JFrog CLI on the JFrog platform, can be automatically generated by the action instead of storing it as a secret in GitHub. 
+| Important: When exposing more than one JFrog configuration to the Action, you should always add the `jf c use` command to specify the server to use. |
+| ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+
+</details>
+
+<details>
+    <summary>Connect using OpenID Connect</summary>
```
Comment (eyalbe4) [inline README.md:132]: Connect using OpenID Connect
-->
Connecting to JFrog using OIDC (OpenID Connect)

File: README.md [hunk 6]
```diff
@@ -9,46 +9,66 @@
...
+## Authorization
+
 JFrog CLI operates in conjunction with the JFrog Platform. In order to facilitate this connection, certain connection details of the JFrog Platform must be provided.
 There exist two methods to provide these details, and you only need to choose **one** method:
-* The connection details of the JFrog platform can be stored as secrets. Read more about it [here](#storing-jfrog-connection-details-as-secrets).
-* The connection details of the JFrog platform can be auto-generated using OpenID Connect protocol. Read more about it [here](#connect-using-openid-connect).
 
+<details>
+    <summary>Storing JFrog connection details as secrets</summary>
+
```
Comment (eyalbe4) [inline README.md:52]: Let's add the following section here, explaining to the user that there are two configuration methods:

### General
You can choose one of the following two methods to set the connection details to the JFrog Platform as secrets.
* Storing the connection details using separate environment variables
* Storing the connection details using single Config Token

File: README.md [hunk 7]
```diff
@@ -59,198 +79,217 @@ You can use these environment variables in your workflow as follows:
...
-## Connect using OpenID Connect
-The sensitive connection details, such as the access token used by JFrog CLI on the JFrog platform, can be automatically generated by the action instead of storing it as a secret in GitHub. 
+| Important: When exposing more than one JFrog configuration to the Action, you should always add the `jf c use` command to specify the server to use. |
+| ---------------------------------------------------------------------------------------------------------------------------------------------------- |
+
+</details>
+
+<details>
+    <summary>Connect using OpenID Connect</summary>
+
```
Comment (eyalbe4) [inline README.md:133]: Let's add a "Geberal" title for the below paragraph, having the same title size as the following "JFrog Platform configuration" title.

File: README.md [hunk 8]
```diff
@@ -9,46 +9,66 @@
...
+-   [Setting JFrog CLI version](#setting-jfrog-cli-version)
+-   [Downloading JFrog CLI from Artifactory](#downloading-jfrog-cli-from-artifactory)
+-   [Set up a FREE JFrog Environment in the Cloud](#set-up-a-free-jfrog-environment-in-the-cloud)
+-   [Example projects](#example-projects)
+-   [Contributions](#contributions)
+-   [References](#references)
+
+## Overview
+
 This GitHub Action downloads, installs and configures JFrog CLI, so that it can be used as part of the workflow.
```
Comment (eyalbe4) [inline README.md:28]: Let's make "JFrog CLI" a link to - 
https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: .github/workflows/oidc-test.yml (hunk 1 above)
Comment (yahavi) [reply .github/workflows/oidc-test.yml:47]: It used to set the "subject" key, but dealing with the "repository" is simpler, so I took it out.
